edx_event_bus_kafka/publishing/event_producer.py

```diff
@@ -0,0 +1,220 @@
+    """
+    return AvroSignalSerializer(signal)
+
+
+# TODO: Cache this, but in a way that still allows changes to settings
+# via remote-config (and in particular does not result in mixed
+# cache/uncached configuration).
+def get_producer_for_signal(signal: OpenEdxPublicSignal, event_key_field: str) -> Optional[SerializingProducer]:
+    """
+    Create the producer for a signal and a key field path.
+
+    If essential settings are missing or invalid, warn and return None.
+
+    Arguments:
+        signal: The OpenEdxPublicSignal to make a producer for
+        event_key_field: Path to the event data field to use as the event key (period-delimited
+          string naming the dictionary keys to descend)
+    """
+    if schema_registry_url := getattr(settings, 'SCHEMA_REGISTRY_URL', None):
+        schema_registry_config = {
+            'url': schema_registry_url,
+            'basic.auth.user.info': f"{getattr(settings, 'SCHEMA_REGISTRY_API_KEY', '')}"
+                                    f":{getattr(settings, 'SCHEMA_REGISTRY_API_SECRET', '')}",
+        }
+    else:
+        warnings.warn("Cannot configure event-bus-kafka: Missing setting SCHEMA_REGISTRY_URL")
+        return None
+
+    if bootstrap_servers := getattr(settings, 'KAFKA_BOOTSTRAP_SERVERS', None):
+        producer_settings = {
+            'bootstrap.servers': bootstrap_servers,
+        }
+    else:
+        warnings.warn("Cannot configure event-bus-kafka: Missing setting KAFKA_BOOTSTRAP_SERVERS")
+        return None
+
+    if getattr(settings, 'KAFKA_API_KEY', None) and getattr(settings, 'KAFKA_API_SECRET', None):
+        producer_settings.update({
+            'sasl.mechanism': 'PLAIN',
+            'security.protocol': 'SASL_SSL',
+            'sasl.username': settings.KAFKA_API_KEY,
+            'sasl.password': settings.KAFKA_API_SECRET,
+        })
+
+    schema_registry_client = SchemaRegistryClient(schema_registry_config)
+    signal_serializer = get_serializer(signal)
+
+    def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
+        """Tells Avro how to turn objects into dictionaries."""
+        return signal_serializer.to_dict(event_data)
+
+    # Serializers for key and value components of Kafka event
+    key_serializer = AvroSerializer(
+        schema_str=extract_key_schema(signal_serializer, event_key_field),
+        schema_registry_client=schema_registry_client,
+        to_dict=inner_to_dict,
+    )
+    value_serializer = AvroSerializer(
+        schema_str=signal_serializer.schema_string(),
+        schema_registry_client=schema_registry_client,
+        to_dict=inner_to_dict,
+    )
+
+    producer_settings.update({
+        'key.serializer': key_serializer,
+        'value.serializer': value_serializer,
+    })
+
+    return SerializingProducer(producer_settings)
+
+
+def verify_event(err, evt):
+    """
+    Simple callback method for debugging event production
+
+    :param err: Error if event production failed
```

Comment:
[nit] Use google style docstring (this is 100% my fault for doing it incorrectly in the original)